### DISCUSSION TITLE
Optional consolidated view for diagnostics

### DISCUSSION THREAD

**Discussion by rafaeldelboni** [clojure-lsp/clojure-lsp#1202] [Ideas]
Refactoring big systems with LSP is incredible and when integrated with your IDE makes the experience even better, but I find the current output of diagnostics to "noisy" when you have this kind of integration.

I would love a way to get an output like this:
```bash
test/unit/file_one_test.clj: infos: 5 warnings: 2 errors: 0
src/namespace/file_one.clj: infos: 2 warnings: 0 errors: 1
```

Because I can check each point in the file using my IDE, this report would fit better to have an overview of whats happening in a big system refactor.

WDYT?

**Comment by ericdallo** [comment]
On emacs this is already possible using the existing `diagnostics` provided by clojure-lsp, maybe you have something similar in your editor? 
![2022-08-19_13-36](https://user-images.githubusercontent.com/7820865/185666306-02eb8004-7ca7-4ddd-a5cc-59196d4a9fa1.png)

**Reply by rafaeldelboni** [reply]
<img width="1220" alt="image" src="https://user-images.githubusercontent.com/1683898/185675477-2c7af9e5-774f-48a6-a06c-9da328ee8644.png">
Yes I do have access to `diagnostics` in my IDE, I was thinking in a report for the CLI, but good point I forgot about the option of using this directly in the IDE 😄

**Reply by ericdallo** [reply]
Oh I thought your intention was to check that on IDE

Right, I think I got your point now, we do have a output on diagnostics, but not grouped by file right?

**Reply by rafaeldelboni** [reply]
Exactly, sorry for not being clear :)

**Reply by ericdallo** [reply]
No problem, I can see how convenient this is, feel free to post a issue to add a new flag to group by file, it shouldn't be hard to implement too
